jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -548,27 +550,28 @@ protected void handleException(Throwable x)
             {
                 // Handle error normally
                 _request.setHandled(true);
-                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION, x);
-                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION_TYPE, x.getClass());
+                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION, failure);
+                _request.setAttribute(RequestDispatcher.ERROR_EXCEPTION_TYPE, failure.getClass());
 
                 if (isCommitted())
                 {
-                    abort(x);
+                    abort(failure);
                     if (LOG.isDebugEnabled())
-                        LOG.debug("Could not send response error 500, already committed", x);
+                        LOG.debug("Could not send response error 500, already committed", failure);
                 }
                 else
                 {
                     _response.setHeader(HttpHeader.CONNECTION.asString(), HttpHeaderValue.CLOSE.asString());
 
-                    if (x instanceof BadMessageException)
+                    Throwable cause = unwrap(failure);
```

Comment:
If you unwrap, don't you have to change the `RequestDispatcher.ERROR_EXCEPTION` and `RequestDispatcher.ERROR_EXCEPTION_TYPE` attributes too?

Reply:
I don't think so.  We are just unwrapping to find our own exceptions, which are either for verbosity or for setting the error code.